src/apps/vpn/platforms/ios/ioscontroller.swift

```diff
@@ -254,11 +256,53 @@ public class IOSControllerImpl : NSObject {
             }
         }
     }
+    
+    /**
+    * @brief Enables / Disables the Always-On Rule on the runnel
+    * This is used for the "start-on-boot" feature. 
+    * We are setting a rule matching * any network change
+    * which will cause the phone to start the VPN after the phone boots.
+    *
+    * @param setEnabled: bool If the rule should be added or removed.
+    **/
+    @objc func setAlwaysOn(setEnabled: Bool) {
+        if (!setEnabled) {
+            tunnel!.isOnDemandEnabled = false;
+            tunnel!.onDemandRules = []
+            return;
+        }
+        let alwaysConnect = NEOnDemandRuleConnect()
+        alwaysConnect.interfaceTypeMatch = .any
+        tunnel!.isOnDemandEnabled = true
+        tunnel!.onDemandRules = [alwaysConnect]
+    }
+    
+    @objc func getAlwaysOn() -> Bool {
+        return tunnel!.isOnDemandEnabled;
+    }
+    
+    @objc func hasAlwaysOn() -> Bool {
```

Comment:
I believe this function will return `false` whenever the user has actively chosen to *not* enable "launch on boot", though we'd want it to return `true`. `onDemandRules` will have 0 rules, and so `isEmpty` will return `true`.

Reply:
>I believe this function will return false whenever the user has actively chosen to not enable "launch on boot"

If the user has actively chosen to not enable "launch on boot" on the Client, that is correct. Given that we clear out the rules when this option is disabled. (Which i think is good, otherwise we can have weird edge-cases)

However, if the user has chosen to disable it on the iOS system settings, the `onDemandRules` will not be empty only `isOnDemandEnabled` will be false. Therefore, I can use it to check if we need to sync the value of `isOnDemandEnabled` back to the client :)

Reply:
I just tested this:
1) In app, activate "always on".
2) In app, deactivate "always on".
3) In iOS Settings, activate "always on".
4) Relaunch app. Our app shows inactive "always on", and iOS settings is updated to also have an inactive "always on".

In this scenario, don't we want our app to match what the user did in iOS settings, and not vice versa?